app/src/main/java/com/wentingzhou/android/fanfouclient/LoginPageActivity.java

```
@@ -57,25 +63,41 @@ public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
         String userAccountPassword = accountInfo.getString(PASSWORDKEY, null);
-        if (accountInfo.getString(USERNAMEKEY, null) == null) {
-            userAccountName =  mUser.getText().toString();
-            userAccountPassword = mPassword.getText().toString();
-        } else {
-            userAccountName = userAccountName + DELIMITER + mUser.getText().toString();
-            userAccountPassword = userAccountPassword + DELIMITER + mPassword.getText().toString();
+        String userAPI = accountInfo.getString(TOKEN, null);
+
+        try {
+            OauthTokenRequest tokenRequest = new OauthTokenRequest();
+            String currentUsername = mUser.getText().toString();
+            String currentPassword = mPassword.getText().toString();
+            tokenRequest.mUsernameInput = currentUsername;
+            tokenRequest.mPasswordInput = currentPassword;
+            FanfouAPI resultAPI = tokenRequest.execute().get();
+            Gson gson = new Gson();
+            String apiJson = gson.toJson(resultAPI.getAccessToken());
+            if (accountInfo.getString(USERNAMEKEY, null) == null) {
+                userAccountName =  currentUsername;
+                userAccountPassword = currentPassword;
+                userAPI = apiJson;
+            } else {
+                userAccountName = userAccountName + DELIMITER + currentUsername;
+                userAccountPassword = userAccountPassword + DELIMITER + currentPassword;
+                userAPI = userAPI + DELIMITER + apiJson;
+            }
+            edit.putString(USERNAMEKEY, userAccountName);
+            edit.putString(PASSWORDKEY, userAccountPassword);
+            edit.putString(TOKEN, userAPI);
+            edit.commit();
+            Intent timeline = new Intent(this, DisplayTimelineActivity.class);
+            timeline.putExtra(DisplayTimelineActivity.USERNAME, currentUsername);
+            startActivity(timeline);
+        } catch (Exception e) {
```

Limit your try block to as little as possible. Otherwise the next time you get an exception, the exception will be swallowed and you will spend hours not knowing what to do. In this case, why do you need to catch all types of exception? And why do you need to catch exception for the entire large block of code above?